### PR TITLE
FB add detekt to gradle

### PR DESCRIPTION
**Pull request by gurps-singh**
This PR is to add detekt for static analysis locally.

### CODE REVIEW

File: detekt.yml [hunk 1]
```diff
@@ -0,0 +1,786 @@
...
+    ignoreEnums: false
+    ignoreRanges: false
+    ignoreExtensionFunctions: true
+  MandatoryBracesLoops:
+    active: false
+  MaxChainedCallsOnSameLine:
+    active: false
+    maxChainedCalls: 5
+  MaxLineLength:
+    active: false
```
Comment (gurps-singh) [inline detekt.yml:633]: We had plenty of `MaxLineLength` errors which we can turn off by setting `active: false`

File: build.gradle.kts [hunk 2]
```diff
@@ -75,10 +76,36 @@ repositories {
...
+  buildUponDefaultConfig = true // preconfigure defaults
+  allRules = false // activate all available (even unstable) rules.
+  config.setFrom("$projectDir/detekt.yml") // point to your custom config defining rules to run, overwriting default behavior
+  baseline = file("$projectDir/detekt-baseline.xml") // a way of suppressing issues before introducing detekt
+}
+
+configurations.matching { it.name == "detekt" }.all {
+  resolutionStrategy.eachDependency {
+    if (requested.group == "org.jetbrains.kotlin") {
+      useVersion("1.8.21")
```
Comment (andrewrlee) [inline build.gradle.kts:90]: do you know why this needed? won't it work with 1.8.22?

File: build.gradle.kts [hunk 3]
```diff
@@ -75,10 +76,36 @@ repositories {
...
       jvmTarget = "18"
     }
   }
+  withType<io.gitlab.arturbosch.detekt.Detekt>().configureEach {
+    reports {
+      html.required.set(true) // observe findings in your browser with structure and code snippets
+      xml.required.set(true) // checkstyle like format mainly for integrations like Jenkins
+      txt.required.set(true) // similar to the console output, contains issue signature to manually edit baseline files
+      sarif.required.set(true) // standardized SARIF format (https://sarifweb.azurewebsites.net/) to support integrations with GitHub Code Scanning
+      md.required.set(true) // simple Markdown format
```
Comment (andrewrlee) [inline build.gradle.kts:107]: if we don't need the others, shall we just switch on html reports for now?

Comment (gurps-singh) [reply build.gradle.kts:107]: Done

File: build.gradle.kts (hunk 2 above)
Comment (gurps-singh) [reply build.gradle.kts:90]: Was running into the behaviour mentioned here in their own documentation and this was their remedy:
[](https://detekt.dev/docs/gettingstarted/gradle/#gradle-runtime-dependencies)

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**